src/components/Modals/NewCourse/RequirementsDropdown.vue

```diff
@@ -0,0 +1,188 @@
+<template>
+  <div v-click-outside="closeDropdownIfOpen">
+    <button
+      class="dropdown"
+      type="button"
+      @click="showDropdown = !showDropdown"
+      data-toggle="dropdown"
+      aria-haspopup="true"
+      :aria-expanded="showDropdown"
+      v-if="potentialRequirements.some(req => req.id === selectedID)"
+    >
+      <div class="warning-row">
+        <img class="warning-icon" src="@/assets/images/warning.svg" alt="warning-icon" />
+        {{ selected }}
+      </div>
+      <drop-down-arrow :isFlipped="showDropdown" :fillColor="emGreen" />
+    </button>
+    <button
+      v-else
+      class="dropdown"
+      type="button"
+      @click="showDropdown = !showDropdown"
+      data-toggle="dropdown"
+      aria-haspopup="true"
+      :aria-expanded="showDropdown"
+    >
+      {{ selected === '' ? 'Select one (optional)' : selected }}
+      <drop-down-arrow :isFlipped="showDropdown" :fillColor="emGreen" />
+    </button>
+    <ul v-if="showDropdown" class="dropdown-content">
+      <li v-for="option in relatedRequirements" :key="option.id">
+        <a
+          @click="toggleSelectRequirement(option.id)"
+          @keyup.enter="toggleSelectRequirement(option.id)"
+          tabindex="0"
+          >{{ option.name }}</a
+        >
+      </li>
+      <li v-for="option in potentialRequirements" :key="option.id">
+        <a
+          @click="toggleSelectRequirement(option.id)"
+          @keyup.enter="toggleSelectRequirement(option.id)"
+          tabindex="0"
+        >
+          <img class="warning-icon" src="@/assets/images/warning.svg" alt="warning-icon" />
+          {{ option.name }}
+        </a>
+      </li>
+      <li>
+        <a
+          @click="toggleSelectRequirement('')"
+          @keyup.enter="toggleSelectRequirement('')"
+          tabindex="0"
+        >
+          None
+        </a>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+
+import { clickOutside } from '@/utilities';
+import DropDownArrow from '@/components/DropDownArrow.vue';
+import { emGreen } from '@/assets/scss/_variables.scss';
```

Comment:
to resolve the type error, add it to the type declaration in `src/assets/scss/_variables.scss.d.ts`